processor/src/decoder/tests.rs

```diff
@@ -1202,7 +1201,7 @@ fn build_trace(stack_inputs: &[u64], program: &CodeBlock) -> (DecoderTrace, usiz
     process.execute_code_block(program, &CodeBlockTable::default()).unwrap();
 
     let (trace, _, _) = ExecutionTrace::test_finalize_trace(process);
-    let trace_len = get_trace_len(&trace) - ExecutionTrace::NUM_RAND_ROWS;
+    let trace_len = trace.num_rows() - ExecutionTrace::NUM_RAND_ROWS;
 
     (
         trace[DECODER_TRACE_RANGE]
```

Comment:
@plafer Thinking about how to handle MainTrace indexed at a range. 
1. Thought of implementing get_column_range(range: Range<usize>) -> &[..] in the MainTrace impl
2. Another option is to implement dedicated methods for retrieving those columns
    - DECODER_TRACE_RANGE
    - SYS_TRACE_RANGE
    - CHIPLETS_RANGE

What do you think? also, is there a better way?